### PR TITLE
not allowed is not unknown

### LINKED ISSUE #2517: trying to spawn with a ckit when it's disabled gives wrong message
It gives: `Unknown starting item`.

For a start, the option should be disabled one way or another, but also the error message should say the item is forbidden, not unknown.
I had this on map `siege`

### PR DESCRIPTION
**Pull request by bmorel**
fix #2517

### CODE REVIEW

File: src/sgame/sg_cmds.cpp [hunk 1]
```diff
@@ -2758,19 +2758,8 @@ static bool Cmd_Class_spawn_internal( gentity_t *ent, const char *s, bool report
...
-			weapon = WP_MACHINEGUN;
-		}
-		else if ( !Q_stricmp( s, BG_Weapon( WP_HBUILD )->name ) &&
-		          !BG_WeaponDisabled( WP_HBUILD ) )
-		{
-			weapon = WP_HBUILD;
-		}
-		else
+		weapon_t weapon = BG_WeaponNumberByName( s );
+		if ( weapon == WP_NONE )
```
Comment (slipher) [inline src/sgame/sg_cmds.cpp:2762]: This allows e.g. `/class lcannon`

Comment (bmorel) [reply src/sgame/sg_cmds.cpp:2765]: oh right. Fixed.

**Comment by bmorel** [comment]
I think this works, I should probably squash the commits though, if someone review?

File: src/sgame/sg_cmds.cpp [hunk 2]
```diff
@@ -2779,6 +2771,12 @@ static bool Cmd_Class_spawn_internal( gentity_t *ent, const char *s, bool report
 			return false;
 		}
 
+		if ( BG_WeaponDisabled( weapon ) || end == std::find( start, end, weapon ) )
+		{
+			G_TriggerMenu( ent->client->ps.clientNum, MN_H_SPAWNITEMNOTALLOWED );
```
Comment (slipher) [inline src/sgame/sg_cmds.cpp:2776]: Needs to respect the `report` option

File: src/sgame/sg_cmds.cpp [hunk 3]
```diff
@@ -2758,19 +2758,11 @@ static bool Cmd_Class_spawn_internal( gentity_t *ent, const char *s, bool report
...
-			weapon = WP_MACHINEGUN;
-		}
-		else if ( !Q_stricmp( s, BG_Weapon( WP_HBUILD )->name ) &&
-		          !BG_WeaponDisabled( WP_HBUILD ) )
-		{
-			weapon = WP_HBUILD;
-		}
-		else
+		weapon_t birthWeapons[] = { WP_HBUILD, WP_MACHINEGUN };
+		auto start = std::begin( birthWeapons );
```
Comment (slipher) [inline src/sgame/sg_cmds.cpp:2762]: Nit: I think it would be more readable without the `start` and `end` variables

**Comment by slipher** [comment]
LGTM

**Pull request merged**